lib/ethclient/ethbackend/backend.go

```diff
@@ -285,6 +286,71 @@ func (b *Backend) EnsureSynced(ctx context.Context) error {
 	return nil
 }
 
+// WaitConfirmed is similar to WaitMined, except that it also waits for the (first) receipt block to be "confirmed".
+// "Confirmed" depends on the evmchain.Reorgs value, if false, it is just calls WaitMined and returns the receipt.
+// Otherwise, it also waits for "safe" confirmation (this is very slow up to 6min).
+// It returns an error if the (first) receipt is reorged out (even though the tx may still be included
+// in a different block).
+func (b *Backend) WaitConfirmed(ctx context.Context, tx *ethtypes.Transaction) (*ethclient.Receipt, error) {
+	rec, err := b.WaitMined(ctx, tx)
+	if err != nil {
+		return nil, errors.Wrap(err, "wait mined")
+	}
+
+	meta, ok := evmchain.MetadataByID(b.chainID)
+	if !ok {
+		return nil, errors.New("unknown chain id", "chain", b.chainName)
```

Comment:
nit: maybe this should be done before waiting for mined?

Reply:
cool, yeah will move